rclrs/src/node/publisher.rs

```diff
@@ -94,17 +95,16 @@ where
         })
     }
 
-    pub fn publish(&self, message: &T) -> Result<(), RclReturnCode> {
```

Comment:
Done, in a sense.

Reply:
What do you mean by "in a sense", are there any caveats?

Reply:
It's not `message: &T` now, it's `message: impl OwnedOrBorrowedMessage`, which accepts `T` and `&T`.

Suggestions for a better name welcome. In case you dislike this design, we could also split it up into two functions, `publish_owned()` and `publish_ref()`.

Reply:
I don't get it, you seemed to be very keen on not having a separate create_subscription function for RMW-compatible types, but `publish_owned` and `publish_ref` are ok. Could you elaborate?

Let's keep it as is now, though, and we could refine the APIs later if needed, though it'll be more painful.

Reply:
Sure. The distinction between `publish_owned()` and `publish_ref()` is of course orthogonal to the question of which message type is published. But you're right that both are a "have two functions for doing the thing vs. having one more general function" question.

In this case here, I'm more on the fence between splitting vs unifying, since the more general function comes at the cost of a more complicated signature: The `MessageCow` trait. In the case of the extra `create_subscription` function, the `create_subscription` function would look the same whether or not we also have a `create_rmw_subscription`, so there's no extra complexity in the unified version. Being simpler is not the only argument in favor of one subscription function, but I'll keep that discussion in the other thread.